src/lib/spinel/spinel.c

```diff
@@ -1634,6 +1634,188 @@ const char *spinel_capability_to_cstr(spinel_capability_t capability)
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
+    case SPINEL_DATATYPE_INT32_C:
+    {
+        int32_t val = ptr_args ? (*va_arg(args, int32_t *)) : va_arg(args, int32_t);
+        bytes       = snprintf(buf, buf_size, "(INT32: %d)", val);
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
+    case SPINEL_DATATYPE_UINT64_C:
+    {
+        uint64_t val = ptr_args ? (*va_arg(args, uint64_t *)) : va_arg(args, uint64_t);
+        bytes        = snprintf(buf, buf_size, "(UINT64: %lu)", val);
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
+    case SPINEL_DATATYPE_INT64_C:
+    {
+        int64_t val = ptr_args ? (*va_arg(args, int64_t *)) : va_arg(args, int64_t);
+        bytes       = snprintf(buf, buf_size, "(INT64: %ld)", val);
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
+    case SPINEL_DATATYPE_IPv6ADDR_C:
+    {
+        spinel_ipv6addr_t *arg_ptr = va_arg(args, spinel_ipv6addr_t *);
+        bytes = snprintf(buf, buf_size, "(IPv6: %x:%x:%x:%x:%x:%x:%x:%x)", (arg_ptr->bytes[0] << 8) | arg_ptr->bytes[1],
+                         (arg_ptr->bytes[2] << 8) | arg_ptr->bytes[3], (arg_ptr->bytes[4] << 8) | arg_ptr->bytes[5],
+                         (arg_ptr->bytes[6] << 8) | arg_ptr->bytes[7], (arg_ptr->bytes[8] << 8) | arg_ptr->bytes[9],
+                         (arg_ptr->bytes[10] << 8) | arg_ptr->bytes[11], (arg_ptr->bytes[12] << 8) | arg_ptr->bytes[13],
+                         (arg_ptr->bytes[14] << 8) | arg_ptr->bytes[15]);
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
+    case SPINEL_DATATYPE_EUI64_C:
+    {
+        spinel_eui64_t *arg_ptr = va_arg(args, spinel_eui64_t *);
+        bytes = snprintf(buf, buf_size, "(EUI64: %x:%x:%x:%x:%x:%x:%x:%x)", arg_ptr->bytes[0], arg_ptr->bytes[1],
+                         arg_ptr->bytes[2], arg_ptr->bytes[3], arg_ptr->bytes[4], arg_ptr->bytes[5], arg_ptr->bytes[6],
+                         arg_ptr->bytes[7]);
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
+    case SPINEL_DATATYPE_EUI48_C:
+    {
+        spinel_eui48_t *arg_ptr = va_arg(args, spinel_eui48_t *);
+        bytes = snprintf(buf, buf_size, "(EUI48: %x:%x:%x:%x:%x:%x)", arg_ptr->bytes[0], arg_ptr->bytes[1],
+                         arg_ptr->bytes[2], arg_ptr->bytes[3], arg_ptr->bytes[4], arg_ptr->bytes[5]);
+        require_action(bytes > 0 && bytes < buf_size, bail, ret = SPINEL_PROP_DISPLAY_ERROR_BUFFER_OVERFLOW);
+        break;
+    }
+
```

Comment:
I guess we can handle more complex formats (structs and array, etc) as well.
For example, for `struct` we can probably use recursive calls to the same function to handle the struct fields.